src/coreclr/vm/jitinterface.cpp

```
@@ -9794,59 +9755,178 @@ CorInfoHFAElemType CEEInfo::getHFAType(CORINFO_CLASS_HANDLE hClass)
+        }
+    }
+
+    CorInfoCallConvExtension getUnmanagedCallConvForMethod(MethodDesc* pMD, bool* pSuppressGCTransition)
+    {
+        ULONG methodCallConv;
+        PCCOR_SIGNATURE pSig;
+        DWORD cbSig;
+        pMD->GetSig(&pSig, &cbSig);
+        if (FAILED(SigParser(pSig, cbSig).GetCallingConv(&methodCallConv)))
+        {
+            COMPlusThrowHR(COR_E_BADIMAGEFORMAT);
+        }
+
+        if (methodCallConv == CORINFO_CALLCONV_DEFAULT || methodCallConv == CORINFO_CALLCONV_VARARG)
+        {
+            _ASSERTE(pMD->IsNDirect() || pMD->HasUnmanagedCallersOnlyAttribute());
+            if (pMD->IsNDirect())
+            {
+                if (pSuppressGCTransition)
+                {
+                    *pSuppressGCTransition = pMD->ShouldSuppressGCTransition();
+                }
+
+                PInvokeStaticSigInfo sigInfo(pMD, PInvokeStaticSigInfo::NO_THROW_ON_ERROR);
+                switch (sigInfo.GetCallConv())
+                {
+                case pmCallConvCdecl:
+                    return CorInfoCallConvExtension::C;
+                    break;
+                case pmCallConvStdcall:
+                    return CorInfoCallConvExtension::Stdcall;
+                    break;
+                case pmCallConvThiscall:
+                    return CorInfoCallConvExtension::Thiscall;
+                    break;
+                case pmCallConvFastcall:
+                    return CorInfoCallConvExtension::Fastcall;
+                    break;
+                default:
+                    _ASSERTE_MSG(false, "bad callconv");
+                    return CorInfoCallConvExtension::Managed;
+                    break;
+                }
+            }
+            else
+            {
+#ifdef CROSSGEN_COMPILE
+                _ASSERTE_MSG(false, "UnmanagedCallersOnly methods are not supported in crossgen and should be rejected before getting here.");
+                return CorInfoCallConvExtension::Managed;
```

Perhaps a fail fast is appropriate?